### PR TITLE
NetworkManager: Store nodes identity directly in `NetworkClient`

### PR DESCRIPTION
**Pull request by Geod24**
One step towards eliminating `NodeConnInfo`.
Part of https://github.com/bosagora/agora/pull/2845

### CODE REVIEW

**Review by omerfirmak — APPROVED** [review]

File: source/agora/network/Client.d [hunk 1]
```diff
@@ -146,6 +146,9 @@ public class NetworkClient
     /// Logger uniquely identifying this client
     private Logger log;
 
+    /// The identity associated with this node - may be empty
+    private Identity identity_;
```
Comment (linked0) [inline source/agora/network/Client.d:157]: Can you explain the meaning of adding `_` in the name other than other member variables?

**Review by linked0 — APPROVED** [review]
LGTM

File: source/agora/network/Client.d (hunk 1 above)
Comment (Geod24) [reply source/agora/network/Client.d:157]: Because otherwise it would conflict with the method that is named `identity`.

**Comment by Geod24** [comment]
Looks to me like the CI failures are not spurious.

**Comment by Geod24** [comment]
So, after some work on this, it's clear this approach here is not going to work because we currently accept any node that present us with a public key as a validator, which requires deeper/more functional changes. Closing and will open other PRs to address the issue.

**Pull request closed without merging**